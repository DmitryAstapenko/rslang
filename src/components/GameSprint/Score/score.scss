.game-sprint-score{
  font-weight: 700;
  font-size: 24px;
  display: flex;
  align-items: center;
}
