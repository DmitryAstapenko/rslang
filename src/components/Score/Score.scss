.level {
  flex-grow: 0.6;
  max-width: 200px;
  margin-right: 1%;
  &__label {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    margin-bottom: 5px;
    &__symbol {
      display: none;
    }
  }
  &__number {
    background-color: white;
    width: 30px;
    height: 30px;
    display: flex;
    justify-content: center;
    color: blue;
    align-items: center;
    font-size: 20px;
    font-weight: 900;
    margin-left: 5px;
  }
}
