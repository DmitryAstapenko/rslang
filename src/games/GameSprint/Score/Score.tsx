import { Typography } from "@material-ui/core";
import React from "react";

interface IScoreProps {
  score: number;
}

const Score: React.FunctionComponent<IScoreProps> = (props: IScoreProps) => {
  const { score } = props;
  return (
    <Typography variant="h5" className="game-sprint-score">
      {`Счёт: ${score}`}
    </Typography>
  );
};

export default Score;
