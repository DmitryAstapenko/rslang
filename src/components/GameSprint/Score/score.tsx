import React from "react";

import "./score.scss";

const Score = () => {
  return <h1>100</h1>;
};

export default Score;
