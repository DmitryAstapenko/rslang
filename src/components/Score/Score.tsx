import React, { FC, useMemo } from "react";
import { Progress } from "react-sweet-progress";
import { Box, LinearProgress } from "@material-ui/core";
import useTypedSelector from "../../hooks/useTypeSelector";
import "react-sweet-progress/lib/style.css";
import "./Score.scss";

const Score: FC = () => {
  const { score } = useTypedSelector((state) => state.score);
  const { loading } = useTypedSelector((state) => state.auth);
  const scorePerLevel = useMemo(() => 300, []);

  const scoreData = {
    levelPercent: ((score % scorePerLevel) * 100) / scorePerLevel,
    level: Math.floor(score / scorePerLevel),
  };

  return (
    <div className="level">
      {loading ? (
        <LinearProgress color="secondary" />
      ) : (
        <>
          <div className="level__label">
            {`Level `}
            <Box borderRadius="50%" className="level__number">
              {scoreData.level}
            </Box>
          </div>
          <Progress
            percent={scoreData.levelPercent}
            symbolClassName="level__label__symbol"
          />
        </>
      )}
    </div>
  );
};

export default Score;
