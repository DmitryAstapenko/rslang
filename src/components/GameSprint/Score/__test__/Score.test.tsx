import React from "react";
import Score from "../Score";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

interface IScoreProps {
  score: number;
}

const setUp = (props:IScoreProps) => shallow(<Score {...props} />);

describe("should render Score component", () => {
  let component:any;
  beforeEach(() => {
    const score: number = 100;
    component = setUp({score});
  });
  it("should contain Score block", () => {
    const wrapper = component.find(".game-sprint-score");
    expect(wrapper.length).toBe(1);
  });
  it("should render Score component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
