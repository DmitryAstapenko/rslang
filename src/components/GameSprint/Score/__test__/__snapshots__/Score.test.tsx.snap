// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Score component should render Score component 1`] = `
<h1
  className="game-sprint-score"
>
  Счёт: 100
</h1>
`;
